src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel_GetCodeWindowInfo.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    internal partial class DocumentOutlineViewModel
+    {
+        /// <summary>
+        /// Queue uses to get the caret position, and data needed to make an LSP request. 
+        /// </summary>
+        private readonly AsyncBatchingResultQueue<DocumentSymbolRequestInfo?> _documentSymbolRequestInfoQueue;
+
+        private async ValueTask<DocumentSymbolRequestInfo?> GetDocumentSymbolRequestInfoAsync(CancellationToken token)
+        {
+            var info = await _visualStudioCodeWindowInfoService.GetDocumentSymbolRequestInfoAsync(token).ConfigureAwait(false);
+            if (info is not null)
+            {
+                _documentSymbolQueue.AddWork(info, cancelExistingWork: true);
+            }
+
+            return info;
```

Comment:
def a bit oogy about both adding into another queue, and returning from this queue... the flow of data and the ordering of queues is not clear here.  needs docs.